src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1548,8 +1557,12 @@ public void pause(final boolean v) throws IOException {
                 if (v) {
                     g.pause();
                     checkAndAbortNonresumableBuild();  // TODO Verify if we can rely on just killing paused builds at shutdown via checkAndAbortNonresumableBuild()
-                    checkpoint();
+                    checkpoint(false);
                 } else {
+                    // Pausing the build sets persistedClean to true so the build can resume, so if we unpause the build
+                    // we need to unset persistedClean again.
+                    persistedClean = null;
+                    saveOwner();
```

Comment:
We should add a test case for this. The Pipeline would need to be paused, then unpaused, and then we do a hard shutdown and make sure we get the error message that the Pipeline can't be resumed rather than a raw exception.